talk/concurrency.tex

```diff
@@ -233,13 +233,43 @@
       int a = 0;
       std::mutex m;
       void inc() {
-        std::lock_guard<std::mutex> guard(m);
+        std::scoped_lock lock{m};
         a++;
       }
     \end{cppcode*}
   \end{exampleblock}
 \end{frame}
 
+\begin{frame}[fragile]
+  \frametitleii{Mutexes and Locks}
+  \begin{block}{Good practice}
+    \begin{itemize}
+      \item Always use \texttt{scoped\_lock} (\cpp11: \texttt{lock\_guard})
+      \item Hold as short as possible, wrap critical section in "\texttt{\string{  \string}}"
+      \item Only if manual control needed, use \texttt{unique\_lock}
```

Comment:
Another use case for that is together with `std::condition_variable`.

Reply:
Good point. We discussed that I'll merge this as the simple version, and there's #62 for the next iteration when we have an advanced course.